src/services/provider/evm/mod.rs

```diff
@@ -118,6 +124,39 @@ impl EvmProvider {
         let provider = ProviderBuilder::new().on_http(rpc_url);
         Ok(Self { provider })
     }
+
+    /// Creates a new EVM provider instance with a custom timeout.
+    ///
+    /// # Arguments
+    /// * `url` - The RPC endpoint URL to connect to
+    /// * `timeout_seconds` - The timeout duration in seconds
+    ///
+    /// # Returns
+    /// * `Result<Self>` - A new provider instance or an error
+    pub fn new_with_timeout(url: &str, timeout_seconds: u64) -> Result<Self> {
+        // Check if URL has valid HTTP(S) scheme because this uses and http client
+        if !url.starts_with("http://") && !url.starts_with("https://") {
```

Comment:
This is probably not needed here but let's leave it for now.

These urls should be validated before we get to this point.